tests/jsunit/xml_test.js

```diff
@@ -75,8 +77,35 @@ function test_domToWorkspace() {
         '  </block>' +
         '</xml>');
     var workspace = new Blockly.Workspace();
-    Blockly.Xml.domToWorkspace(dom, workspace);
+	Blockly.Xml.domToWorkspace(dom, workspace);
     assertEquals('Block count', 1, workspace.getAllBlocks().length);
+	Blockly.Xml.domToWorkspace(dom, workspace,[50,50]);
+	assertEquals('Block count', 2, workspace.getAllBlocks().length);
+  } finally {
+    delete Blockly.Blocks.test_block;
+  }
+}
+
+function test_appendDomToWorkspace() {
+  Blockly.Blocks.test_block = {
+    init: function() {
+      this.jsonInit({
+        message0: 'test',
+      });
+    }
+  };
+
+  try {
+    var dom = Blockly.Xml.textToDom(
+        '<xml xmlns="http://www.w3.org/1999/xhtml">' +
+        '  <block type="test_block" inline="true" x="21" y="23">' +
+        '  </block>' +
+        '</xml>');
+    var workspaceA = new Blockly.Workspace();
```

Comment:
Unfortunately, our test coverage of this web version of Blockly is pretty sparse, especially for things that are rendered, like any of the SVG classes.

WorkspaceSVG @extends Blockly.Workspace, and Blockly.mainWorkspace is almost always a WorkspaceSVG object.  There are several examples of WorkspaceSVG.getCanvas() and .getBubbleCanvas() in the WorkspaceSVG class itself.  What problem were you having?